src/overlays/actors/ovl_Bg_F40_Swlift/z_bg_f40_swlift.c

```diff
@@ -28,24 +31,81 @@ ActorInit Bg_F40_Swlift_InitVars = {
+static InitChainEntry sInitChain[] = {
     ICHAIN_F32(uncullZoneScale, 550, ICHAIN_CONTINUE),
     ICHAIN_F32(uncullZoneForward, 5000, ICHAIN_CONTINUE),
     ICHAIN_VEC3F_DIV1000(scale, 100, ICHAIN_STOP),
 };
 
-#endif
+void BgF40Swlift_Init(Actor* thisx, PlayState* play) {
+    s32 index;
+    BgF40Swlift* this = THIS;
+
+    Actor_ProcessInitChain(&this->dyna.actor, sInitChain);
+    DynaPolyActor_Init((DynaPolyActor*)this, 1);
+    index = BG_F40_GET_SWITCHFLAG(this);
+    if ((index < 0) || (index >= 5)) { //! @bug An index greater than 3 will cause an out of bounds array access.
+        Actor_Kill(&this->dyna.actor);
+    } else {
+        sHeights[index] = this->dyna.actor.world.pos.y;
+        sParams[index] = BG_F40_SWLIFT_GET_PARAMS(this);
+        if (index) {
+            Actor_Kill(&this->dyna.actor);
+        } else {
+            DynaPolyActor_LoadMesh(play, &this->dyna, &gUnusedStoneTowerVerticallyOscillatingPlatformCol);
+            this->dyna.actor.params = 0;
+        }
+    }
+}
+
+void BgF40Swlift_Destroy(Actor* thisx, PlayState* play) {
+    BgF40Swlift* this = THIS;
+    DynaPoly_DeleteBgActor(play, &play->colCtx.dyna, this->dyna.bgId);
+}
 
-extern InitChainEntry D_8096F540[];
+void BgF40Swlift_Update(Actor* thisx, PlayState* play2) {
+    PlayState* play = play2;
+    BgF40Swlift* this = THIS;
+    s32 i;
 
-extern UNK_TYPE D_06003B08;
-extern UNK_TYPE D_06003E80;
+    for (i = 1; i < ARRAY_COUNT(sParams); i++) {
+        if ((sParams[i] == 0xFF) || (Flags_GetSwitch(play, sParams[i]) == 0)) {
+            break;
+        }
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_F40_Swlift/BgF40Swlift_Init.s")
+    i--;
+    if (i != this->dyna.actor.params) {
+        f32 heightOffset;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_F40_Swlift/BgF40Swlift_Destroy.s")
+        this->dyna.actor.params = -1;
+        heightOffset = (((f32)sHeights[i]) - this->dyna.actor.world.pos.y) * 0.1f;
+        if (heightOffset > 0.0f) {
+            heightOffset = CLAMP(heightOffset, 0.5f, 15.0f);
+        } else {
+            heightOffset = CLAMP(heightOffset, -15.0f, -0.5f);
+        }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_F40_Swlift/BgF40Swlift_Update.s")
+        if ((Math_StepToF(&this->dyna.actor.speedXZ, heightOffset, 1.0f) != 0) && (fabsf(heightOffset) <= 0.5f)) {
+            if (Math_StepToF(&this->dyna.actor.world.pos.y, (f32)sHeights[i], fabsf(this->dyna.actor.speedXZ)) != 0) {
+                this->dyna.actor.params = i;
+                this->timer = 48;
+                this->dyna.actor.speedXZ = 0.0f;
+            }
+        } else {
+            this->dyna.actor.world.pos.y += this->dyna.actor.speedXZ;
+        }
+    } else {
+        if (this->timer == 0) {
+            this->timer = 48;
+        }
+        this->timer--;
+        this->dyna.actor.world.pos.y =
+            sHeights[this->dyna.actor.params] + (sin_rad(((f32)this->timer) * (M_PI / 24.0f)) * 5.0f);
```

Comment:
```suggestion
            sHeights[this->dyna.actor.params] + (sin_rad(this->timer * (M_PI / 24.0f)) * 5.0f);
```
Is the cast required?